src/bdc/steps/regionalatlas.py

```diff
@@ -61,87 +62,122 @@ def verify(self) -> bool:
-
-        country = google_location[-1].lower()
+        # can only get an result if we know the region
+        if (
+            row["google_places_formatted_address"] is None
+            and row["number_area"] is None
+        ):
+            return self.empty_result
+
+        country = ""
+
+        # the phone number has secondary priority (because it can be a private number), therefore can be overwritten by the google places information
+        if row["number_country"] is not None:
+            country = row["number_country"]
+
+        if row["google_places_formatted_address"] is not None:
+            google_location = str(row["google_places_formatted_address"]).split(",")[
+                -2:
+            ]
+            google_location = [name.strip() for name in google_location]
+            country = google_location[-1].lower()
 
         # the 'regionalatlas' data is specific to germany
-        if country not in ['deutschland','germany','allemagne','tyskland','germania']:
-            return empty_result
-        
-        # Alternative to the if 'if country not in ...'
-        # if not self.germany_gdf.intersects(row_gdf):
-        #     return empty_result
+        if country not in [
+            "deutschland",
+            "germany",
+            "allemagne",
+            "tyskland",
+            "germania",
+        ]:
+            return self.empty_result
+
+        """#Alternative to the if 'if country not in ...'
+        if not self.germany_gdf.intersects(row_gdf):
+            return self.empty_result"""
 
         # Get the polygon of the city, to find the corresponding region
         try:
-            search_gdf = osmnx.geocode_to_gdf(",".join(google_location))            
+            if row["google_places_formatted_address"] is not None:
+                search_gdf = osmnx.geocode_to_gdf(",".join(google_location))
+            else:  # at this point we know, that either a google_places_address exists or a number_area
+                search_gdf = osmnx.geocode_to_gdf(row["number_area"])
         except:
-            log.info('Google location not found!')
-            return empty_result
-        
-        # Load the data file
-        #!!!!! INTO THE RUN PART
-        try:
-            regions_gdfs = gpd.read_file("data/merged_geo.geojson") #{"gdp": gpd.read_file("data/merged_geo.geojson")}
-        except:            
-            log.info("File does not exist!")
-            return empty_result
-                
-        # WGS 84 (used by osmnx)
-        epsg_code_ll = 4326
+            log.info("Google location not found!")
+            return self.empty_result
 
-        #ETRS89 / UTM zone 32N (used by regionalatlas)
+        # Adjust the EPSG code from the osmnx search query to the regionalatlas specific code
+        # epsg_code 4326 [WGS 84 (used by osmnx)]=> epsg_code_etrs = 25832 [ETRS89 / UTM zone 32N (used by regionalatlas)]
         epsg_code_etrs = 25832
```

Comment:
Maybe this magic number can be defined as a class variable for all instances of the regionalatlas step

Reply:
True, good idea